src/azul/openapi/validation.py

```diff
@@ -0,0 +1,141 @@
+            if type(filter_json) is not dict:
+                raise BadRequestError(f'The {param_name} parameter must be a dictionary')
+            else:
+                for facet, filter_ in filter_json.items():
+                    try:
+                        relation, value = one(filter_.items())
+                    except Exception:
+                        raise BadRequestError(
+                            f"The 'filters' parameter entry for {filter_!r} must be a single-item"
+                            f" dictionary")
+                    try:
+                        jsonschema.validate({facet: filter_}, self.schema)
+                    except (
+                        jsonschema.exceptions.SchemaError,
+                        jsonschema.exceptions.ValidationError
+                    ) as e:
+                        if (
+                            len(e.absolute_schema_path) == 1
+                            and 'additionalProperties' in e.absolute_schema_path
+                        ):
+                            raise BadRequestError(f'Unknown facet {facet!r}')
+                        elif 'oneOf' and 'additionalProperties' in e.absolute_schema_path:
+                            valid_relations = [
+                                one(list(relations['properties'].keys())) for relations in
+                                e.parent.schema['oneOf']
+                            ]
+                            raise BadRequestError(f'The relation in the {param_name!r} parameter entry'
+                                                  f' for {facet!r}. Must be one of {valid_relations!r}.')
+                        elif 'minItems' or 'maxItems' in e.absolute_schema_path:
+                            relation_spec = one(relation_ for relation_ in e.parent.schema['oneOf']
+                                                if relation in relation_['properties'])
+                            raise BadRequestError(msg=f'The value of the {relation!r} relation in the '
+                                                      f'{param_name!r} parameter entry for {facet!r} does not match '
+                                                      f'the spec {relation_spec!r}')
+                        else:
+                            raise BadRequestError(e)
+        else:
+            try:
+                self.validator(param_value)
+            except (TypeError, ValueError):
+                raise BadRequestError(msg=f'Invalid value for parameter {param_name!r}')
+            assert self.schema is not None
+            if self.schema.get('format') == 'uuid':
+                def uuid_validator(validator, value, instance, schema):
+                    try:
+                        validate_uuid(instance)
+                    except (InvalidUUIDError, InvalidUUIDVersionError):
+                        yield jsonschema.exceptions.ValidationError('%r is an invalid UUID' % instance)
```

Comment:
I'm super confused by this yield statement. Normally we just raise exception instances. I didn't see anything in the jsonschema docs about this (but I didn't look too hard). What's going on here?

Reply:
i recall reading in the jsonschema documentation that a yield statement should be used. I cant locate it at the moment, and upon inspecting https://python-jsonschema.readthedocs.io/en/latest/errors/?#jsonschema.exceptions.ValidationError a return or raise can be used.

Reply:
https://github.com/Julian/jsonschema/blob/master/jsonschema/_validators.py

Reply:
not using a yield statement fails to set information about the exception being raised, this causes issues with figuring out what caused the issue.